src/ol/expr/expression.js

```diff
@@ -91,6 +91,8 @@ import {toSize} from '../size.js';
  *   * `['!', value1]` returns `false` if `value1` is `true` or greater than `0`, or `true` otherwise.
  *   * `['all', value1, value2, ...]` returns `true` if all the inputs are `true`, `false` otherwise.
  *   * `['any', value1, value2, ...]` returns `true` if any of the inputs are `true`, `false` otherwise.
+ *   * `['has', attributeName, keyOrArrayIndex, ...]` returns `true` if feature properties include the (nested) key `attributeName`,
+ *     `false` otherwise.
```

Comment:
```suggestion
 *     `false` otherwise.  If the property exists but the value is `null` or `undefined`, the expression evaluates to `false`.
```

This behavior might be different than what people expect (different from [`Object.hasOwn()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn) for example).  I think it should be documented at least.

Reply:
@tschaub I hadn't thought of this, you are right. Initially, I wanted to name the operation `defined`, but @jahow pointed out the `has` API and we decided to name it this way. But you are right, the mapbox `has` API tests `...for a presence of a property`, the big question being, if `undefined` or `null` count as present (same as they do with `hasOwn`).

I'm open to reverting to my initial idea of naming it `defined`, which would be a more exact description of the current expression. Additionally, if you see the use case, introduce a second operation `has` and use `Object.hasOwn` to test for `presence`.

Reply:
@bkuster - I don't think things would be clearer with `defined`.  For example, I can define a property on an object whose value is `undefined` (`{example: undefined}` or `Object.defineProperty({}, 'example', {value: undefined})`).

I think the semantics of `has` are easier to guess/understand.  The documentation that you included in your commit describes `has` well.  I pushed another commit that updates the implementation to match the documentation.